src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -541,11 +538,9 @@ private void getMatchingAttributes() throws IOException {
                 attributeInfos = connection.getAttributesForBean(beanName);
             } catch (IOException e) {
                 // we should not continue
-                log.warn("Cannot get bean attributes or class name: " + e.getMessage());
-                if (e.getMessage().equals(connection.CLOSED_CLIENT_CAUSE)) {
-                    throw e;
-                }
-                continue;
+                log.warn("[IOException] Cannot get bean attributes or class name: {}",
+                    e.getMessage());
+                throw e;
```

Comment:
When we throw here it does produce a traceback from [this bit](https://github.com/DataDog/jmxfetch/blob/cdaa75d4caa245ac8f018e7f34548ab56469fb68/src/main/java/org/datadog/jmxfetch/App.java#L1159-L1166) of `App.java`, so the warning log is un-necessary.
With this warning log the logs when connection breaks during attribute matching looks like this:
```
2023-04-12 17:15:25 UTC | JMX | INFO | Instance | Refreshing bean list
2023-04-12 17:15:34 UTC | JMX | WARN | JmxAttribute | Unable to get metrics from test:type=MyMBean,name=Example_0000006250 - FloatValue: java.rmi.UnmarshalException: Error unmarshaling return header; nested exception is:
        java.net.SocketException: Connection reset
2023-04-12 17:15:34 UTC | JMX | WARN | JmxAttribute | Unable to get metrics from test:type=MyMBean,name=Example_0000006250 - BooleanValue: java.rmi.ConnectException: Connection refused to host: 172.17.0.2; nested exception is:
        java.net.ConnectException: Connection refused
2023-04-12 17:15:34 UTC | JMX | WARN | JmxAttribute | Unable to get metrics from test:type=MyMBean,name=Example_0000006250 - DoubleValue: java.rmi.ConnectException: Connection refused to host: 172.17.0.2; nested exception is:
        java.net.ConnectException: Connection refused
2023-04-12 17:15:34 UTC | JMX | WARN | Instance | [IOException] Cannot get bean attributes or class name: Connection refused to host: 172.17.0.2; nested exception is:
        java.net.ConnectException: Connection refused
2023-04-12 17:15:34 UTC | JMX | WARN | App | Unable to collect metrics or refresh bean list.
java.rmi.ConnectException: Connection refused to host: 172.17.0.2; nested exception is:
        java.net.ConnectException: Connection refused
        at java.rmi/sun.rmi.transport.tcp.TCPEndpoint.newSocket(TCPEndpoint.java:626)
        at java.rmi/sun.rmi.transport.tcp.TCPChannel.createConnection(TCPChannel.java:217)
        at java.rmi/sun.rmi.transport.tcp.TCPChannel.newConnection(TCPChannel.java:204)
        at java.rmi/sun.rmi.server.UnicastRef.invoke(UnicastRef.java:134)
        at jdk.remoteref/jdk.jmx.remote.internal.rmi.PRef.invoke(Unknown Source)
        at java.management.rmi/javax.management.remote.rmi.RMIConnectionImpl_Stub.getMBeanInfo(RMIConnectionImpl_Stub.java:381)
        at java.management.rmi/javax.management.remote.rmi.RMIConnector$RemoteMBeanServerConnection.getMBeanInfo(RMIConnector.java:1078)
        at org.datadog.jmxfetch.Connection.getClassNameForBean(Connection.java:49)
        at org.datadog.jmxfetch.Instance.getMatchingAttributes(Instance.java:534)
        at org.datadog.jmxfetch.Instance.getMetrics(Instance.java:458)
        at org.datadog.jmxfetch.MetricCollectionTask.call(MetricCollectionTask.java:26)
        at org.datadog.jmxfetch.MetricCollectionTask.call(MetricCollectionTask.java:8)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: java.net.ConnectException: Connection refused
        at java.base/sun.nio.ch.Net.connect0(Native Method)
        at java.base/sun.nio.ch.Net.connect(Net.java:579)
        at java.base/sun.nio.ch.Net.connect(Net.java:568)
        at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:576)
        at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
        at java.base/java.net.Socket.connect(Socket.java:666)
        at java.base/java.net.Socket.connect(Socket.java:600)
        at java.base/java.net.Socket.<init>(Socket.java:509)
        at java.base/java.net.Socket.<init>(Socket.java:289)
        at java.rmi/sun.rmi.transport.tcp.TCPDirectSocketFactory.createSocket(TCPDirectSocketFactory.java:40)
        at java.rmi/sun.rmi.transport.tcp.TCPEndpoint.newSocket(TCPEndpoint.java:620)
        ... 15 more
```

I'll update this to remove the `log.warn`, but we need the catch block to re-throw this exception, otherwise the below `catch` would handle this and simply log a warning and continue, which is not what we want.